### PR TITLE
B 19926 loa lookup UI int

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

Introduces long line of accounting display and concatenation on the orders page for office users. Includes a warning for missing / invalid TAC.

The state of the LOA preview should change when:

- The TAC is updated and is equal to a length of 4
- The issue date changes
- The department changes

### How to test

1. Log into milmovelocal
2. Create a new move and HHG shipment
3. Write down the move locator
4. Log into officelocal as a service counselor
5. Navigate to the move -> edit orders
6. Fill out the orders like so:
  1. Issue date: 2014-01-01
  2. Department: ARMY
  3. TAC: E12A
7. Upon filling out the above, the LOA preview at the bottom should populate (Subject to change based on randomly generated seed data)
8. Save the form if you see the LOA preview populate
9. Switch to TIO/TOO
10. Navigate to the move
11. Open edit orders -> the LOA should populate successfully without you touching anything

## Screenshots

image
 
image

image

### CODE REVIEW

**Comment by cameroncaci** [comment]
flaky tests

File: src/components/Office/OrdersDetailForm/OrdersDetailForm.jsx [hunk 1]
```diff
@@ -139,6 +143,22 @@ const OrdersDetailForm = ({
...
+          id="loaTextField"
+          mask="****"
+          inputTestId="loaTextField"
+          data-testid="loaTextField"
+          warning={loaWarning}
+          validate={validateLOA}
+          value={longLineOfAccounting}
+          isDisabled
+        />
+      )}
```
Comment (danieljordan-caci) [inline src/components/Office/OrdersDetailForm/OrdersDetailForm.jsx:160]: Couldn't you do:
```
      {showLoa && (
        <h3>Line of Accounting Preview</h3>
        <TextField
          name="loa"
          label="LOA"
          id="loaTextField"
          mask="****"
          inputTestId="loaTextField"
          data-testid="loaTextField"
          warning={loaWarning}
          validate={validateLOA}
          value={longLineOfAccounting}
          isDisabled
        />
      )}

Comment (cameroncaci) [reply src/components/Office/OrdersDetailForm/OrdersDetailForm.jsx:160]: When I initially tried that the linter got mad at me.. hmm lemme try again

Comment (cameroncaci) [reply src/components/Office/OrdersDetailForm/OrdersDetailForm.jsx:160]: Yeah, it gives this: `JSX expressions must have one parent element.ts(2657)`

**Review by TevinAdams — APPROVED** [review]

**Pull request merged**